File: app/src/main/java/com/example/dataLayer/interfaces/PostRepositoryInterface.java
package com.example.dataLayer.interfaces;

import com.example.dataLayer.models.PostDTO;

import java.util.ArrayList;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;

public interface PostRepositoryInterface {
    @GET("RestfulRequestHandler.php")
    Call<ArrayList<PostDTO>> fetchPosts(@Query("recentPosts") boolean recentPosts);


    @GET("RestfulRequestHandler.php")
    Call<ArrayList<PostDTO>> fetchNewPosts(@Query("recentPosts") boolean recentPosts, @Query("lastPostID") long lastPostID);

    @GET("RestfulRequestHandler.php")
    Call<PostDTO> fetchPostByID(@Query("postID") long postID);

    @GET("RestfulRequestHandler.php")
    Call<ArrayList<PostDTO>> fetchFavoritePostsByUserID(@Query("userID") String userID, @Query("savedPosts") boolean favoritePosts);

    @GET("RestfulRequestHandler.php")
    Call<ArrayList<PostDTO>> fetchMyPosts(@Query("userID") String userID, @Query("myPosts") boolean favoritePosts);


    @POST("RestfulRequestHandler.php")
    Call<PostDTO> uploadPost(@Query("uploadPost") boolean uploadPost, @Body PostDTO serializePost);


    @FormUrlEncoded
    @POST("RestfulRequestHandler.php")
    Call<String> addPostToFavorites(@Query("addPostToFavorite") boolean addToFavorites, @Field("postID") long postID, @Field("userID") String userID);

    @FormUrlEncoded
    @POST("RestfulRequestHandler.php")
    Call<PostDTO> removePostsFromFavorites(@Query("addPostToFavorite") boolean addToFavorites, @Field("postID") int postID, @Field("userID") String userID);


}

File: app/src/main/java/com/example/dataLayer/interfaces/PostRepositoryInterface.kt
package com.example.dataLayer.interfaces

import com.example.bookapp.models.LowDataPost
import com.example.dataLayer.models.PostDTO
import com.example.dataLayer.models.ServerResponse
import com.example.dataLayer.models.SerializeImage
import com.example.dataLayer.models.serialization.SerializePost
import retrofit2.http.*

interface PostRepositoryInterface {

    @GET("/recentPosts")
    suspend fun fetchRecentPosts(): ArrayList<PostDTO>

    @GET("/posts/page/{lastPostID}")
    suspend fun fetchNextPagePosts(@Path("lastPostID")lastPostID: Int): List<PostDTO>


    @GET("/post/{id}")
    suspend fun fetchPostByID(@Path("id") postID: Int): PostDTO

    @GET("/user/{id}/favoritePosts")
    suspend fun fetchUserFavoritePosts(@Path("id") userID: Int): ArrayList<PostDTO>


    @GET("/user/{id}/posts")
    suspend fun fetchMyPosts(@Path("id") userID: Int): ArrayList<PostDTO>

    @DELETE("/user/{userID}/removePost/{postID}")
    suspend fun removePostFromFavorites(@Path("userID") userID: Int, @Path("postID") postID: Int): ServerResponse

    @POST("/user/{userID}/addToFavorites/{postID}")
    suspend fun addPostToFavorites(@Path("postID") postID: Int, @Path("userID") userID: Int)

    @POST("/posts/uploadImage")
    suspend fun uploadImage(@Body serializeImage: SerializeImage): ServerResponse

    @POST("/posts/create")
    suspend fun uploadPost(@Body uploadPost: SerializePost): ServerResponse


}